homeassistant/helpers/integration_platform.py

```diff
@@ -138,16 +186,52 @@ async def async_process_integration_platforms(
     integration_platform = IntegrationPlatform(
         platform_name, process_job, top_level_components
     )
+    # Tell the loader that it should try to pre-load the integration
+    # for any future components that are loaded so we can reduce the
+    # amount of import executor usage.
+    async_register_preload_platform(hass, platform_name)
     integration_platforms.append(integration_platform)
-
     if not top_level_components:
         return
 
     integrations = await async_get_integrations(hass, top_level_components)
-    if futures := [
-        future
-        for comp in top_level_components
-        if (platform := _get_platform(integrations[comp], comp, platform_name))
-        and (future := hass.async_run_hass_job(process_job, hass, comp, platform))
-    ]:
+    loaded_integrations: list[Integration] = []
+    for domain, integration in integrations.items():
+        if isinstance(integration, Exception):
+            _LOGGER.exception(
```

Comment:
This shouldn't happen right? We are fetching integrations of components that have already been set up (as they were in `hass.config.components`

Reply:
Right it should be impossible. We have a bug in setup if it hits this path.

Reply:
21ab42078837be9d08415965c86a297ff32a48a6